tests/parametertree/test_parametertypes.py

```diff
@@ -163,6 +164,17 @@ def override():
     pi.widget.setValue(2)
     assert p.value() == pi.widget.value() == 1
 
+
+def test_checklist_show_hide():
+    p = pt.Parameter.create(name='checklist', type='checklist', limits=["a", "b", "c"])
+    pi = ChecklistParameterItem(p, 0)
+    pi.setHidden = MagicMock()
+    p.hide()
+    pi.setHidden.assert_called_with(True)
+    p.show()
+    pi.setHidden.assert_called_with(False)
+
```

Comment:
I recommend also adding a check here that `p.opts["visible"]` matches expectations, since that's the serialized state